tabbycat/draw/manager.py

```diff
@@ -220,7 +265,16 @@ def get_teams(self):
                 setattr(team, pullup_metric, standing.metrics[pullup_metric])
             ranked.append(team)
 
-        return ranked
+        n_byes = self.n_byes(len(ranked))
+        if n_byes:
+            if self.round.tournament.pref('bye_team_selection') == 'random':
+                byes = []
+                for i in range(n_byes):
+                    byes.append(ranked.pop(random.randrange(len(ranked))))
+                return ranked, byes
+            return ranked[:-n_byes], ranked[-n_byes:]
```

Comment:
I'd recommend writing the `elif` case explicitly and raising an error to fail fast if the preference isn't recognised (so that if in future someone adds a fourth option they don't miss this spot here).